itou/www/accounts/views.py

```diff
@@ -0,0 +1,73 @@
+from allauth.account import app_settings
+from allauth.account.models import EmailAddress, EmailConfirmationHMAC
+from allauth.account.utils import send_email_confirmation
+from allauth.account.views import ConfirmEmailView as BaseConfirmEmailView
+from django.conf import settings
+from django.core import signing
+from django.http import Http404
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.utils import timezone
+from django.views.generic import View
+
+from itou.emails.models import Email
+from itou.utils.auth import LoginNotRequiredMixin
+
+
+class ExpiredConfirmationMixin:
+    def get_object_with_expired_key(self):
+        key = self.kwargs["key"]
+        try:
+            pk = signing.loads(key, salt=app_settings.SALT)
+            return EmailConfirmationHMAC(EmailAddress.objects.get(pk=pk, verified=False)).email_address
+        except (
+            signing.SignatureExpired,
+            signing.BadSignature,
+            EmailAddress.DoesNotExist,
+        ):
+            raise Http404()
+
+
+class ConfirmEmailView(BaseConfirmEmailView, ExpiredConfirmationMixin):
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.object is None:
+            try:
+                self.get_object_with_expired_key()  # Raises error for invalid keys.
+                key = self.kwargs["key"]
+            except Http404:
+                key = None
+        context.update({"provided_key": key})
```

Comment:
Plus clair et plus performant.
```suggestion
        context["provided_key"] = key
```

Reply:
Bien sûr 😅 merci